TombEngine/Scripting/Internal/TEN/Objects/Moveable/MoveableObject.cpp

```diff
@@ -1016,3 +1047,104 @@ bool Moveable::MeshExists(int index) const
 
 	return true;
 }
+
+//Attach camera and camera target to a mesh of an object.
+void  Moveable::AttachCamera(short camMeshId, short targetMeshId)
+{
+	ObjCamera(m_item, camMeshId, m_item, targetMeshId, true);
+}
+
+//Lara will undraw her weapons if they are drawn, throw away flare if she helds one in her hand.
+void   Moveable::UndrawWeapons()
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+
+		if (lara->Control.HandStatus != HandStatus::Free ||
+			lara->Control.Weapon.GunType == LaraWeaponType::Flare)
+		{
+			lara->Control.HandStatus = HandStatus::WeaponUndraw;
+		}
+	}
+}
+
+//Lara will throw away the torch if she helds one in her hand.
+void   Moveable::ThrowawayTorch()
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+
+		if (lara->Control.Weapon.GunType == LaraWeaponType::Torch)
+		{
+			Lara.Torch.State = TorchState::Dropping;
+		}
+	}
+}
+
+//Returns 0=HandsFree, 1=Busy(climbing,etc), 2=WeaponDraw, 3=WeaponUndraw, 4=WeaponInHand.
+HandStatus Moveable::GetLaraHandStatus() const
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+		return  HandStatus{ lara->Control.HandStatus };
+	}
+}
+
+//Returns 0=None, 1=Pistols, 2=Revolver, 3=Uzi, 4=Shotgun, 5=HK, 6=Crossbow, 7=Flare, 8=Torch, 9=GrenadeLauncher, 10=Harpoon, 11=RocketLauncher.
+LaraWeaponType Moveable::GetLaraWeaponType() const
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+		return  LaraWeaponType{ lara->Control.Weapon.GunType };
+	}
+}
+
+//Borrow an animtaion and state id from an object.
+void Moveable::AnimFromObject(GAME_OBJECT_ID object, int animNumber, int stateID)
+{
+	m_item->Animation.AnimNumber = Objects[object].animIndex + animNumber;
+	m_item->Animation.ActiveState = stateID;
+	m_item->Animation.FrameNumber = g_Level.Anims[m_item->Animation.AnimNumber].frameBase;
+	AnimateItem(m_item);
+}
+
+//Set Lara Weapon type.
+//0=None, 1=Pistols, 2=Revolver, 3=Uzi, 4=Shotgun, 5=HK, 6=Crossbow, 7=Flare, 8=Torch, 9=GrenadeLauncher, 10=Harpoon, 11=RocketLauncher.
+void Moveable::SetLaraWeaponType(LaraWeaponType weaponType, bool activate)
+{
+	if (m_item->IsLara())
+	{
+		auto* lara = GetLaraInfo(m_item);
+
+		switch (weaponType)
+		{
+		case LaraWeaponType::Flare:
+			lara->Control.Weapon.RequestGunType = LaraWeaponType::Flare;
+		break;
+		case LaraWeaponType::Torch:
+			using namespace TEN::Entities::Generic;
+			if (activate == false)
+			{
+				GetFlameTorch();
+				lara->Torch.IsLit = false;
+				break;
```

Comment:
Why `break` is placed inside condition here (I mean, inside curly brackets)? You need just a single `break` at the end of the case, outside of curly brackets.